src/main/java/org/spongepowered/api/event/impl/AbstractModifierEvent.java

```diff
@@ -56,20 +56,24 @@
     protected ImmutableList<T> init(double originalValue, List<T> originalFunctions) {
         final ImmutableList.Builder<Tuple<M, Double>> modifierMapBuilder = ImmutableList.builder();
         final ImmutableList.Builder<T> functionListBuilder = ImmutableList.builder();
-        final ImmutableMap.Builder<M, Double> mapBuilder = ImmutableMap.builder();
+        final Map<M, Double> mapBuilder = new LinkedHashMap<>();
         double finalDamage = originalValue;
         for (T tuple : originalFunctions) {
             this.modifierFunctions.add(convertTuple(tuple.getModifier(), tuple.getFunction()));
             double tempDamage = checkNotNull(tuple.getFunction().applyAsDouble(finalDamage));
             finalDamage += tempDamage;
             modifierMapBuilder.add(new Tuple<>(tuple.getModifier(), tempDamage));
-            mapBuilder.put(tuple.getModifier(), tempDamage);
+            if (mapBuilder.containsKey(tuple.getModifier())) {
+                mapBuilder.put(tuple.getModifier(), mapBuilder.get(tuple.getModifier()) + tempDamage);
+            } else {
+                mapBuilder.put(tuple.getModifier(), tempDamage);
+            }
```

Comment:
Isn't it possible to simplyfy this like:

`mapBuilder.merge(tuple.getModifier(), tempDamage, (a,b) -> a + b);`

[`Map#merge(K key, V value, BiFunction<? super V,? super V,? extends V> remappingFunction)`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#merge-K-V-java.util.function.BiFunction-)

Reply:
I'll do that.